### PR TITLE
feat: fill functionality in config editor and validation of config files

### PR DESCRIPTION
**Pull request by Dexnjn**
# Content
This pull request contains the addition of a "fill" popover for the data_editor and validation for the config file in the ace_editor view as well as some minor fixes.

## Release Notes

- **New Features**
  - Introduced a new configuration editor with ACE editor integration for enhanced editing capabilities.
  - Added a data fill feature to streamline populating configuration files from selected datasets.

- **Improvements**
  - Enhanced metadata handling in various components for better organization and consistency.
  - Updated session state management for improved user experience and validation processes.

- **Bug Fixes**
  - Resolved inconsistencies in session state key formats and validation logic across multiple components.

### CODE REVIEW

**Review by johanneskoester — APPROVED** [review]

**Comment by johanneskoester** [comment]
@coderabbitai please review

**Comment by johanneskoester** [comment]
@coderabbitai review

**Comment by johanneskoester** [comment]
Let us wait for the AI review. From my side this looks good.

File: common/components/config_editor.py [hunk 1]
```diff
@@ -59,37 +93,57 @@ def create_form(config: dict, schema: dict, wd, parent_key: str = "") -> dict:
...
             unique_element_id = update_key(parent_key, key)
             input_dict = schema[prop_key].get(key)
+            if not ace_editor or (
+                input_dict["type"] == "string"
+                and value
+                and value.endswith((".tsv", ".csv", ".xlsx"))
+            ):
+                only_validation = False
+            else:
+                only_validation = True
```
[the remark this answers is not in the document]
Comment (Dexnjn) [reply common/components/config_editor.py]: Clearer this way 👍🏻

File: common/components/data_editor.py [hunk 2]
```diff
@@ -1,462 +1,597 @@
...
+    key : str
+        The key for the Streamlit session state of the data.
+    """
+    editor = st.session_state[key + "-editor"]["edited_rows"]
+    data = st.session_state[key + "-data"]
+    coldictkey = list(editor.keys())[0]
+    colname = list(editor[coldictkey].keys())[0]
+    data.loc[coldictkey, colname] = editor[coldictkey][colname]
+    st.session_state[key + "-data"] = data
+
```
[the remark this answers is not in the document]
Comment (Dexnjn) [reply common/components/data_editor.py]: I could not trigger the warning through the interface but adding error handling is likely not a bad choice

File: common/components/config_editor.py [hunk 3]
```diff
@@ -105,69 +159,82 @@ def get_input_element(
...
+                        st.empty() for _ in range(3)
+                    ]
+                    if show_data & isinstance(st.session_state[data_key], DataFrame):
+                        data_editor(st.session_state[data_key], key)
+            case input_type if input_type == "integer":
+                input_value = st.number_input(label=label, value=value, key=key)
+            case input_type if input_type == "number":
+                if "e" in str(value).lower():
+                    input_value = st.text_input(label=label, value=value, key=key)
+                else:
```
[the remark this answers is not in the document]
Comment (Dexnjn) [reply common/components/config_editor.py:209]: This should be caught in validate_input to keep generation and validation separate.

File: common/components/data_editor.py [hunk 4]
```diff
@@ -1,462 +1,597 @@
...
+            local_vars[local_var_key] = value
+    if dataframe_type == "Polars":
+        data = pl.from_pandas(data)
+    local_vars["df"] = data
+    exec(user_code, {}, local_vars)
+    data = local_vars.get("df", data)
+    if not isinstance(data, pd.DataFrame):
+        data = data.to_pandas()
+    return data
+
```
[the remark this answers is not in the document]
Comment (Dexnjn) [reply common/components/data_editor.py:309]: We handle this by disabling imports

File: common/components/ui_components.py [hunk 5]
```diff
@@ -18,7 +18,7 @@ def toggle_button(label: str, key: str) -> bool:
     def toggle(state):
         st.session_state[state] = not st.session_state[state]
 
-    unique_key = label.lower() + key
+    unique_key = key + "-" + label.lower()
```
Comment (johanneskoester) [inline common/components/ui_components.py]:
```suggestion
    unique_key = f"{key}-{label.lower()}"
```

**Review by johanneskoester — CHANGES_REQUESTED** [review]

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**